Framework/Crystal/src/CentroidPeaks.cpp

```diff
@@ -355,5 +363,32 @@ void CentroidPeaks::removeEdgePeaks(
   peakWS.removePeaks(std::move(badPeaks));
 }
 
+void CentroidPeaks::sizeBanks(std::string bankName, int &nCols, int &nRows) {
+  if (bankName == "None")
+    return;
+  boost::shared_ptr<const IComponent> parent =
+      inst->getComponentByName(bankName);
+  if (!parent)
+    return;
+  if (parent->type() == "RectangularDetector") {
```

Comment:
Unfortuantely #20274 isn't quite ready yet, but that would have helped a lot as that provides a `isRectangularBank` method. i,e you could have written.

```cpp
auto index = componentInfo.indexOf(parent->getComponentID());
if(componentInfo.isRectangularBank(index)){
....
}
```
And avoided the string comparison. For the time being, would it be better to define/use a `static const std::string RectangularDetectorType` defined somewhere rather than a temporary string for this comparison?

Reply:
Removed